### PR TITLE
Disable most CustomCaST system test cases for KRaft mode

### PR DESCRIPTION
**Pull request by see-quick**
Signed-off-by: see-quick <maros.orsak159@gmail.com>

### Type of change

- Bugfix

### Description

This PR fixes the KRaft pipeline, which is currently failing due to test cases that retrieve ZK secrets (which is impossible in KRaft mode). I have disabled such test cases, and in the following PR, I will change the logic of test cases, which will also support KRaft mode.

### Checklist

- [x] Make sure all tests pass

### CODE REVIEW

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java [hunk 1]
```diff
@@ -68,6 +69,7 @@ public class CustomCaST extends AbstractST {
     private static final String STRIMZI_INTERMEDIATE_CA = "C=CZ, L=Prague, O=Strimzi, CN=StrimziIntermediateCA";
 
     @ParallelNamespaceTest
+    @KRaftNotSupported
```
Comment (Frawless) [inline systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java:72]: Description why it is not supported would be nice

**Review by Frawless — APPROVED** [review]
Just a nit, you can add the same description as you have in PR description.

**Review by scholzj — APPROVED** [review]
Agree with Jakub's comment. It would be good to explain the reasons directly in the code. LGTM otherwise.

File: systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java (hunk 1 above)
Comment (see-quick) [reply systemtest/src/test/java/io/strimzi/systemtest/security/custom/CustomCaST.java]: Done.

**Comment by see-quick** [comment]
The KRaft pipeline has one failure not related to this PR (flaky test case, which I have fixed in this PR (https://github.com/strimzi/strimzi-kafka-operator/pull/8077). 

image

**Pull request merged**